script/zip-comp-logs

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+"""
+A script to zip the logs produced by a series of competition matches.
+"""
+
+import re
+import shutil
+import argparse
+import datetime
+import contextlib
+from typing import Union, Iterator
+from pathlib import Path
+from zipfile import ZipFile
+
+
+def zip_folder(folder: Path, args: argparse.Namespace) -> str:
```

Comment:
As a general pattern, I would encourage that `args` only be handled at the top level and that any relevant values are passed explicitly into utils. Note also that `argparse.Namespace` isn't usefully typed, so using it here doesn't offer any safety.